swaybar/status_line.c

```diff
@@ -115,6 +115,14 @@ bool status_handle_readable(struct status_line *status) {
 		sway_log(SWAY_DEBUG, "Using text protocol.");
 		status->protocol = PROTOCOL_TEXT;
 		status->text = status->buffer;
+		newline = strrchr(status->buffer, '\n');
+		if (newline != NULL && newline[1] == '\0') {
+			newline[0] = '\0';
+			newline = strrchr(status->buffer, '\n');
```

Comment:
If there is a single newline (e.g. "line 1 \n lin", with the second line cut off), `strrchr` will return NULL. I think we want to fallback to `status->buffer` here.

Reply:
Hm. In case of partial reads the whole `getline` logic below will be broken anyways… I think to fix it properly we need to replace the logic above in `PROTOCOL_UNDEF` with a `getline` call.

Reply:
Tried this and it broke the i3bar protocol parser, likely because `FILE*` has an internal buffer and the i3bar parser uses the FD directly. Oh well.